ui/app/pages/swaps/view-quote/view-quote.js

```diff
@@ -114,6 +115,17 @@ export default function ViewQuote() {
   const usedQuote = selectedQuote || topQuote
   const tradeValue = usedQuote?.trade?.value ?? '0x0'
 
+  const shouldShowPriceDifferenceWarning = ['high', 'medium'].includes(
+    usedQuote?.priceSlippage?.bucket,
+  )
+  let priceDifferencePercentage = 0
+  if (usedQuote?.priceSlippage?.ratio) {
+    priceDifferencePercentage = parseFloat(
+      ((usedQuote.priceSlippage.ratio - 1) * 100).toFixed(2),
```

Comment:
This should probably be done with BigNumber, to avoid risk of rounding or other number errors. I think the following should work:
```
priceDifferencePercentage = (new BigNumber(usedQuote.priceSlippage.ratio, 10))
    .minus(1, 10)
    .times(100, 10)
    .toFixed(2)
```